be2-scala/src/test/scala/ch/epfl/pop/pubsub/graph/validator/SchemaValidatorSuite.scala

```diff
@@ -0,0 +1,62 @@
+package ch.epfl.pop.pubsub.graph.validator
+
+import org.scalatest.{Matchers, FunSuite}
+import scala.io.Source
+import ch.epfl.pop.pubsub.graph.Validator._
+import ch.epfl.pop.pubsub.graph.PipelineError
+import ch.epfl.pop.pubsub.graph.ErrorCodes
+import java.io.IOException
+
+object SchemaValidatorSuite extends FunSuite with Matchers {
+
+  def getJsonStringFromFile(filePath: String): String = {
+      val source = Source.fromFile(filePath);
+      val jsonStr = source.getLines.mkString
+      source.close
+      jsonStr
+    }
+
+  /* Test subscribe query without message format*/
+  test("Correct subscribe JSON-RPC query"){
+    val subscribePath = "../protocol/examples/query/subscribe/subscribe.json"
+    val subscribeJson = getJsonStringFromFile(subscribePath)
+
+    validateSchema(subscribeJson) should be (Left(subscribeJson))
+  }
+
+  test("Incorrect subscribe query: additional params"){
+    val subscribePath = "../protocol/examples/query/subscribe/wrong_subscribe__additional_params.json"
+    val subscribeJson = getJsonStringFromFile(subscribePath)
+
+    validateSchema(subscribeJson) shouldBe a [Right[_,PipelineError]]
+  }
```

Comment:
Nice start! Glanced over the file real quick and noticed that you should probably check that the pipeline error is indeed what we expect (i.e. correct error number or indeed contains an id)